RadixWallet/Core/SharedModels/LedgerIdentifiable.swift

```diff
@@ -33,7 +33,11 @@ public enum LedgerIdentifiable: Sendable {
 // MARK: LedgerIdentifiable.Address
 extension LedgerIdentifiable {
 	public enum Address: Hashable, Sendable, Identifiable {
-		case account(AccountAddress, isLedgerHWAccount: Bool = false)
+		/// `isLedgerHWAccount` indicates if the account is controlled by a Ledger device.
+		/// - `true`: we know the account is controlled by a Ledger device
+		/// - `false`: either the account isn't controller by a Ledger device or it is an external account and we don't care.
+		/// - `nil`: we don't know if the account is controller by a Ledger device, we should check if needed.
+		case account(AccountAddress, isLedgerHWAccount: Bool?)
```

Comment:
`isLedgerHWAccount` complicates things more than it helps, and by having it passed from outside of AddressDetails just saves a very quick check in the Profile(even if it is async), additionally very likely that at some point somebody will miss-use this. For the sake of consistency and encapsulation, I would completely remove this `isLedgerHWAccount`, unless there is a use case which I am missing.

Reply:
definitely, I had left it with the optional case as I thought it was used somewhere else, but seems the only usage of it is for this button. Agree 100% that's better to do the quick async check every time.